src/cli-validator/utils/validator.js

```diff
@@ -60,14 +60,21 @@ module.exports = function validateSwagger(allSpecs, config) {
   }
 
   // run semantic validators
-  const allValidators = Object.assign(
-    {},
-    semanticValidators,
-    sharedSemanticValidators
-  );
 
-  Object.keys(allValidators).forEach(key => {
-    const problem = allValidators[key].validate(allSpecs, config);
+  Object.keys(semanticValidators).forEach(key => {
+    const problem = semanticValidators[key].validate(allSpecs, config);
+    if (problem.errors.length) {
+      validationResults.errors[key] = [...problem.errors];
+      validationResults.error = true;
+    }
+    if (problem.warnings.length) {
+      validationResults.warnings[key] = [...problem.warnings];
+      validationResults.warning = true;
+    }
+  });
+
+  Object.keys(sharedSemanticValidators).forEach(key => {
+    const problem = sharedSemanticValidators[key].validate(allSpecs, config);
     if (problem.errors.length) {
       validationResults.errors[key] = [...problem.errors];
```

Comment:
Now that this has been pointed out, I actually think this is still buggy. If an error was reported by a semantic validator with key `foo` and then also by a shared validator with key `foo`, the first errors would be overwritten, right? Maybe we need to rethink this a little more

Reply:
I see that now.  Just pushed a fix to handle this case.